packages/runtime/src/execute/expression.ts

```diff
@@ -100,13 +101,30 @@ export const wrapOperation = (
   fn: Operation,
   logger: Logger,
   name: string,
-  immutableState?: boolean
+  immutableState?: boolean,
+  prevName?: string
 ) => {
   return async (state: State) => {
     logger.debug(`Starting operation ${name}`);
     const start = new Date().getTime();
+    if (checkAndClearNullState(state)) {
+      logger.warn(`Operation ${name} might fail!`);
+      logger.warn(
+        `The previous operation ${prevName} didn't return a state. did you forget?`
+      );
+    }
     const newState = immutableState ? clone(state) : state;
-    const result = await fn(newState);
+
+    if (typeof fn !== 'function') {
```

Comment:
@josephjclark 
This function check is all we need!

1. it makes sure that fn can be called. prevent error. skipping operation
2. this also works for async functions, because, they are functions that basically return a promise.
3. this also takes care of unwanted top-level code. 🥳

#### Takes care of unwanted top-level code. 
This is because the `fn` value we're validating here isn't directly the function call at the top-level but rather what that function call returns. 

So if you have eg.
```js
map("$.[*]",
  create("SObject",
    field("FirstName", sourceValue("$.firstName"))
  )
)
```
then that `fn` check in our code is for the function returned by our top-level code `map`. 

Note: All of our top-level functions are high-order functions with some signature
```js
function map(...args){
	return (state) => {
		// do some work here
	}
}
```

Hence checking whether `fn` is a function is actually checking whether some top-level code is a high-order function. which most unwanted top-level code fail to satisfy. 
eg. 
`console.log("")`
`setTimeout()`
...